Formula/clusterawsadm.rb

```diff
@@ -0,0 +1,31 @@
+class Clusterawsadm < Formula
+  desc "CLI tool which provides bootstrapping, AMI, EKS, and controller related helpers in Cluster API Provider for AWS"
+  homepage "https://cluster-api-aws.sigs.k8s.io/clusterawsadm/clusterawsadm.html"
+  url "https://github.com/kubernetes-sigs/cluster-api-provider-aws.git",
+      tag:      "v2.0.2",
+      revision: "28bc9b8756d6d7f73038604038ddb3ccf4b22396"
+  license "Apache-2.0"
+  head "https://github.com/kubernetes-sigs/cluster-api-provider-aws.git", branch: "main"
+
+  # Upstream creates releases on GitHub for the two most recent major/minor
+  # versions (e.g., 0.3.x, 0.4.x), so the "latest" release can be incorrect. We
+  # don't check the Git tags for this project because a version may not be
+  # considered released until the GitHub release is created.
+  livecheck do
+    url "https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases"
+   regex(%r{href=["']?[^"' >]*?/tag/v?(\d+(?:\.\d+)+)["' >]}i)
+    strategy :page_match
+  end
+
+  depends_on "go" => :build
+
+  def install
+    # Don't dirty the git tree
+    rm_rf ".brew_home"
+
+    system "make", "clusterawsadm"
+    prefix.install "bin"
+
+    generate_completions_from_executable(bin/"clusterawsadm", "completion", shells: [:bash, :zsh])
+  end
+end
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Let me know if this looks ok, actually all the commands' pre-requisite is to create a cluster using cluster API provider AWS, which won't be possible to add. So added command line validation failure as a test.